client/network/src/block_request_handler.rs

```diff
@@ -24,19 +24,33 @@ use crate::protocol::{message::BlockAttributes};
 use crate::request_responses::{IncomingRequest, OutgoingResponse, ProtocolConfig};
 use crate::schema::v1::block_request::FromBlock;
 use crate::schema::v1::{BlockResponse, Direction};
+use crate::{PeerId, ReputationChange};
 use futures::channel::{mpsc, oneshot};
 use futures::stream::StreamExt;
 use log::debug;
+use lru::LruCache;
 use prost::Message;
 use sp_runtime::generic::BlockId;
 use sp_runtime::traits::{Block as BlockT, Header, One, Zero};
 use std::cmp::min;
-use std::sync::{Arc};
+use std::sync::Arc;
 use std::time::Duration;
+use std::hash::{Hasher, Hash};
 
 const LOG_TARGET: &str = "block-request-handler";
 const MAX_BLOCKS_IN_RESPONSE: usize = 128;
 const MAX_BODY_BYTES: usize = 8 * 1024 * 1024;
+const MAX_NUMBER_OF_SAME_REQUESTS_PER_PEER: usize = 2;
+
+mod rep {
+	use super::ReputationChange as Rep;
+
+	/// Reputation change when a peer sent us an invalid request.
+	pub const INVALID_REQUEST: Rep = Rep::new(-(1 << 16), "Invalid block request");
+
+	/// Reputation change when a peer sent us the same request multiple times.
+	pub const SAME_REQUEST: Rep = Rep::new(i32::min_value(), "Same block request multiple times");
```

Comment:
Why is it so punitive to ask the same thing twice?

Reply:
I'm still waiting for some answer from the author of the linked issue to see if that fixes the problem. As reported, they assume that someone is spamming them with the same request over and over again. 

It is also fine to send the same request twice (or even more times when you wait long enough between the requests), however the same request should not be send multiple times in a short time period. This probably just reveals some bug in the sync code, but I still don't see any problem why we should no restrict this.